doc/js_tutorials/js_assets/js_image_classification.html

```diff
@@ -0,0 +1,384 @@
+                                <td id="prob2" align="center"></td>
+                        </tr>
+                    </tbody>
+                </table>
+                <p id='time' align="left"></p>
+            </td>
+        </tr>
+        <tr>
+            <td>
+                <div class="caption">
+                    canvasInput <input type="file" id="fileInput" name="file" accept="image/*">
+                </div>
+            </td>
+            <td></td>
+        </tr>
+    </table>
+</div>
+
+<div>
+    <h3>Help function</h3>
+    <p>1.The parameters for model inference which you can modify to investigate more models.</p>
+    <textarea class="code" rows="15" cols="100" id="codeEditor" spellcheck="false"></textarea>
+    <p>2.Main loop in which will read the image from canvas and do inference once.</p>
+    <textarea class="code" rows="27" cols="100" id="codeEditor1" spellcheck="false"></textarea>
+    <p>3.Load labels from txt file and process it into an array.</p>
+    <textarea class="code" rows="9" cols="100" id="codeEditor2" spellcheck="false"></textarea>
+    <p>4.Get blob from image as input for net, and standardize it with <b>mean</b> and <b>std</b>.</p>
+    <textarea class="code" rows="11" cols="100" id="codeEditor3" spellcheck="false"></textarea>
+    <p>5.Fetch model from network and save to emscripten file system.</p>
+    <textarea class="code" rows="35" cols="100" id="codeEditor4" spellcheck="false"></textarea>
+    <p>6.The post-processing, including softmax if needed and get the top classes from the output vector.</p>
+    <textarea class="code" rows="35" cols="100" id="codeEditor5" spellcheck="false"></textarea>
+    <p class="err" id="errorMessage"></p>
+</div>
+
+<div id="appendix">
+    <h2>Model Info:</h2>
+</div>
+
+<script src="utils.js" type="text/javascript"></script>
+
+<script id="codeSnippet" type="text/code-snippet">
+inputSize = [224,224];
+mean = [104, 117, 123];
+std = 1;
+rgb = true;
+
+// record if need softmax function for post-processing
+needSoftmax = false; 
+
+// url for model, config and label files, can from native or Internet
+// for model type like onnx which doesn't need config file, just let configUrl = ""
+labelsUrl = "labels1000.txt";
+modelUrl = "bvlc_googlenet.caffemodel";
+configUrl = "bvlc_googlenet.prototxt";
+</script>
+
+<script id="codeSnippet1" type="text/code-snippet">
+main = async function() {
+    const url = {
+        modelUrl: modelUrl,
+        configUrl: configUrl
+    }; 
+    const path = await loadModel(url);
+    const labels = await loadLables(labelsUrl);
+    const input = getBlodFromImage(inputSize, mean, std, rgb);
+    let net;
+    if(path.configPath == "") {
+        net = cv.readNet(path.modelPath);
+    } else {
+        net = cv.readNet(path.configPath, path.modelPath);
+    }
+    net.setInput(input);
+    const start = performance.now();
+    const result = net.forward();
+    const time  = performance.now()-start;
+    const probs = softmax(result);
```

Comment:
SoftMax is not increasing function that means `softmax(x) > softmax(y) <=> x > y`. So to compute only indices we don't need it. Except you need the probabilities as well.

Reply:
Softmax functionn is necessary to me since I need to show the probabilities in the webpage.

Reply:
Got it, thanks!